low-level/framework/utils/consulstore.py

```diff
@@ -144,9 +220,27 @@ def get_keys_with_prefix(self, prefix):
             except requests.exceptions.ConnectionError as connerr:
                 logger.warn("Error[{0}] consul connection refused Retry Index {1}" \
                     .format(connerr, retry_index))
+                self._consul_conn_status = False
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
                 logger.warn("Error[{0}] while getting keys with given prefix {1}" \
                     .format(gerr, prefix))
+                self._consul_conn_status = False
                 break
+
+    def get_keys_with_prefix(self, prefix):
```

Comment:
add some code comments for each stage in this function

Reply:
done